app/src/main/java/org/mifos/selfserviceapp/ui/activities/LoginActivity.java

```diff
@@ -62,8 +73,24 @@ public void onLoginSuccess(String userName) {
         showToast(getString(R.string.toast_welcome, userName));
         llLogin.setVisibility(View.GONE);
         loginPresenter.loadClient();
-    }
+        if (!usernames.contains(userName)) {
+            usernames.add(userName);
+            setAutoCompleteSource(userName);
 
+        }
+    }
+    private void saveprfs(String userName) {
+        SharedPreferences settings = getSharedPreferences(PREFS_NAME, 0);
```

Comment:
Please use `api/local/PreferencesHelper` instead of making new SharedPreferences.